Tests/Editor/Selection/SelectFacesByColor.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UObject = UnityEngine.Object;
+using NUnit.Framework;
+using UnityEngine.ProBuilder;
+using UnityEditor.ProBuilder;
+using UnityEditor;
+using UnityEngine.TestTools;
+
+
+namespace UnityEngine.ProBuilder.EditorTests.Selection
+{
+    class SelectFacesByColor
+    {
+        ProBuilderMesh[] selectables;
+        List<Color> faceColors;
+
+        void Setup(bool withColor = false)
+        {
+            ProBuilderMesh shape1 = ShapeGenerator.CreateShape(ShapeType.Cube);
+            shape1.transform.position = Vector3.zero - shape1.GetComponent<MeshRenderer>().bounds.center;
+
+            ProBuilderMesh shape2 = ShapeGenerator.CreateShape(ShapeType.Cube);
+            shape2.transform.position = Vector3.zero - shape2.GetComponent<MeshRenderer>().bounds.center;
+
+            if (withColor)
+            {
+                //Cube has 6 faces
+                faceColors = new List<Color>();
+                faceColors.Add(Color.red);
+                faceColors.Add(Color.green);
+                faceColors.Add(Color.blue);
+                faceColors.Add(Color.grey);
+                faceColors.Add(Color.cyan);
+                faceColors.Add(Color.black);
+
+                for(var i = 0; i < 6; ++i)
+                {
+                    shape1.SetFaceColor(shape1.facesInternal[i], faceColors[i]);
+                }
+
+                for (var i = 0; i < 6; ++i)
+                {
+                    shape2.SetFaceColor(shape2.facesInternal[i], faceColors[i]);
+                }                
+            }
+
+            selectables = new ProBuilderMesh[]
+            {
+                shape1,
+                shape2
+            };
+        }
+
+        void Cleanup()
```

Comment:
If you add [TearDown] attribute this will automatically happen after each test, whether or not the test fails.

Reply:
Makes sense! Thank you!